### PR TITLE
chore: cleanup

### PR DESCRIPTION
**Pull request by DavidMinarsch**

### CODE REVIEW

File: contracts/BuyBackBurner.sol [hunk 1]
```diff
@@ -102,17 +102,14 @@ contract BuyBackBurner {
...
     function checkPoolPrices(
-        address nativeToken,
-        address memeToken,
+        address token0,
+        address token1,
         address uniV3PositionManager,
-        uint24 fee,
-        bool isNativeFirst
+        uint24 fee
     ) external {
```
Comment (DavidMinarsch) [inline contracts/BuyBackBurner.sol:109]: This contract should be as generic as possible so we can reuse it elsewhere, moved the sorting out

File: contracts/MemeBase.sol [hunk 2]
```diff
@@ -68,21 +67,23 @@ contract MemeBase is MemeFactory {
...
-    /// Hearters of the original AGNT now have 24 hours to collect their `AGNT II`.
+    /// LP setup (same amount of AGNT II and ETH, modulus a possibly rounding difference 
+    /// as new MemeBase uses UniV3, rather than UniV2), as the original AGN has.
+    /// Hearters of the original AGNT now have 24 hours to collect their `AGNT II` allocation.
     function _MAGA() private {
         // Get meme summon info
         MemeSummon storage memeSummon = memeSummons[launchCampaignNonce];
 
         // Unleash the token
-        _unleashThisMeme(launchCampaignNonce, memeSummon, LIQUIDITY_AGNT, CONTRIBUTION_AGNT, 0);
```
Comment (DavidMinarsch) [inline contracts/MemeBase.sol:81]: Explained below

File: contracts/MemeFactory.sol [hunk 3]
```diff
@@ -6,8 +6,7 @@ import {Meme} from "./Meme.sol";
 import {IUniswapV3} from "./interfaces/IUniswapV3.sol";
 
 interface IBuyBackBurner {
-    function checkPoolPrices(address nativeToken, address memeToken, address uniV3PositionManager, uint24 fee,
-        bool isNativeFirst) external;
+    function checkPoolPrices(address token0, address token1, address uniV3PositionManager, uint24 fee) external;
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:9]: cleaned up the interface

File: contracts/MemeFactory.sol [hunk 4]
```diff
@@ -59,12 +62,12 @@ interface IERC20 {
 ///           $SMTH to be burned.
 abstract contract MemeFactory {
     event OLASJourneyToAscendance(uint256 amount);
-    event Summoned(address indexed summoner, uint256 indexed memeNonce, uint256 nativeTokenContributed);
+    event Summoned(address indexed summoner, uint256 indexed memeNonce, uint256 amount);
     event Hearted(address indexed hearter, uint256 indexed memeNonce, uint256 amount);
-    event Unleashed(address indexed unleasher, address indexed memeToken, uint256 indexed lpTokenId,
-        uint256 liquidity, uint256  nativeAmountForOLASBurn);
+    event Unleashed(address indexed unleasher, uint256 indexed memeNonce, address indexed memeToken, uint256 lpTokenId,
+        uint256 liquidity);
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:68]: Due to how the uniV3 dust works, it makes no sense returning this value. The on-chain value will be different potentially, just causing for confusion...

File: contracts/MemeFactory.sol [hunk 5]
```diff
@@ -228,8 +231,10 @@ abstract contract MemeFactory {
     /// @param positionId LP position ID.
     /// @param isNativeFirst Order of a native token in the pool.
     function _collectFees(address memeToken, uint256 positionId, bool isNativeFirst) internal {
+        (address token0, address token1) = isNativeFirst ? (nativeToken, memeToken) : (memeToken, nativeToken);
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:234]: Moving this here so BuyBackBurner is clean

File: contracts/MemeFactory.sol [hunk 6]
```diff
@@ -301,7 +311,7 @@ abstract contract MemeFactory {
         bytes memory payload = abi.encodePacked(type(Meme).creationCode, abi.encode(name, symbol, DECIMALS, totalSupply));
         // solhint-disable-next-line no-inline-assembly
         assembly {
-            memeToken := create2(0x0, add(0x20, payload), mload(payload), memeNonce) // revert to randomNonce
+            memeToken := create2(0x0, add(0x20, payload), mload(payload), memeNonce) // TOFIX: revert to randomNonce
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:314]: Will have to be done as the last PR when we did all tests

File: contracts/MemeFactory.sol [hunk 7]
```diff
@@ -350,16 +362,13 @@ abstract contract MemeFactory {
         memeTokens.push(memeToken);
         numTokens = memeTokens.length;
 
-        // Record msg.sender activity
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:353]: Should not be at nested level, moved up to caller

File: contracts/MemeFactory.sol [hunk 8]
```diff
@@ -59,12 +62,12 @@ interface IERC20 {
 ///           $SMTH to be burned.
 abstract contract MemeFactory {
     event OLASJourneyToAscendance(uint256 amount);
-    event Summoned(address indexed summoner, uint256 indexed memeNonce, uint256 nativeTokenContributed);
+    event Summoned(address indexed summoner, uint256 indexed memeNonce, uint256 amount);
     event Hearted(address indexed hearter, uint256 indexed memeNonce, uint256 amount);
-    event Unleashed(address indexed unleasher, address indexed memeToken, uint256 indexed lpTokenId,
-        uint256 liquidity, uint256  nativeAmountForOLASBurn);
+    event Unleashed(address indexed unleasher, uint256 indexed memeNonce, address indexed memeToken, uint256 lpTokenId,
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:67]: Returning nonce here allows for simpler off-chain correlation of nonce and token

File: contracts/MemeFactory.sol [hunk 9]
```diff
@@ -447,10 +455,10 @@ abstract contract MemeFactory {
         MemeSummon storage memeSummon = memeSummons[memeNonce];
 
         // Check if the meme has been summoned
+        require(memeSummon.summonTime > 0, "Meme not yet summoned");
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:458]: Just cleaned up the order for symmetry

File: contracts/MemeFactory.sol [hunk 10]
```diff
@@ -467,10 +475,12 @@ abstract contract MemeFactory {
         uint256 nativeAmountForLP = totalNativeTokenCommitted - nativeAmountForOLASBurn;
 
         // Schedule native token amount for ascendance
-        // uint256 adjustedNativeAmountForAscendance = _updateLaunchCampaignBalance(nativeAmountForOLASBurn);
         scheduledForAscendance += nativeAmountForOLASBurn;
 
-        _unleashThisMeme(memeNonce, memeSummon, nativeAmountForLP, totalNativeTokenCommitted, nativeAmountForOLASBurn);
+        // Record msg.sender activity
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:480]: Moved up here from lower level

File: contracts/MemeFactory.sol [hunk 11]
```diff
@@ -542,25 +552,25 @@ abstract contract MemeFactory {
...
+        // This condition means launchCampaign can only be triggered once there's a positive
+        // amount for ascendance remaining.
         require(amount > 0, "Nothing to send");
 
         scheduledForAscendance = 0;
 
         // Record msg.sender activity
         mapAccountActivities[msg.sender]++;
 
-        if (amount > 0) {
```
Comment (DavidMinarsch) [inline contracts/MemeFactory.sol:558]: This can never be hit as we already check for amount>0 above

File: contracts/MemeBase.sol [hunk 12]
```diff
@@ -96,6 +97,9 @@ contract MemeBase is MemeFactory {
         _launched = 1;
     }
 
+    /// @dev Native token amount to wrap.
+    /// @notice Celo's native token CELO is also an ERC20.
```
Comment (DavidMinarsch) [inline contracts/MemeBase.sol:101]:
```suggestion
```

**Review by kupermind — APPROVED** [review]

**Pull request merged**